src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -195,6 +197,45 @@ public TutorialClass findTutorialClassFromList(TutorialClass tutorialClass, Modu
         return tutorialClassInList;
     }
 
+    /**
+     * Randomly allocates the students in {@code tutorial class} into {@code numOfTeams} of different teams.
+     *
+     * @param moduleCode of the tutorial class.
+     * @param tutorialClass to allocate the different students into the teams to.
+     * @param numOfTeams of teams to split into.
+     */
+    public void randomTeamAllocation(ModuleCode moduleCode, TutorialClass tutorialClass, int numOfTeams) {
```

Comment:
I think this random allocation could produce very imbalanced teams if unlucky
E.g. 16 people for 5 teams -> math.ceil gives 4 per team.
Thus its possible that you might get 4 teams of 4 and a team of 0 right if I read this right

Reply:
Perhaps a better way would be to assign a student to each team position (i.e. choose a random not yet chosen student to be team 1 person 1, then for team 2 person 1, and so on). That way the team size for each team would be more or less even.

Reply:
this makes sense. I will look into a fix in due course. But for now I think for small cases of class size this is sufficient.

Reply:
Thanks for bringing up, very good spot!